yt_idefix/data_structures.py

```diff
@@ -724,3 +797,241 @@ def _validate_units_override_keys(cls, units_override):
             )
 
         super(cls, cls)._validate_units_override_keys(units_override)
+
+
+class PlutoXdmfDataset(PlutoVtkDataset):
+    _index_class = PlutoXdmfHierarchy
+    _field_info_class = PlutoXdmfFields
+    _dataset_type = "pluto-xdmf"
+    _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
+    _required_header_keyword = "PLUTOXdmf"
+    _default_definitions_header = "definitions.h"
+    _default_inifile = "pluto.ini"
+    _field_offset_index = None
+
+    def _parse_parameter_file(self):
+        grid_file = os.path.join(os.path.dirname(self.parameter_filename), "grid.out")
+        self.parameter_filename[:-2] + "xmf"
+        out_file = os.path.join(
+            os.path.dirname(self.parameter_filename),
+            self.parameter_filename[-6:] + ".out",
```

Comment:
This gets rid of few of the extensions at the end. So I thought directly slicing is good. However, let me think of something better.

Reply:
consider using `str.partition` to split all extensions at once in a more expressive and robust way
Here's a simple illustration of what it does
```python
>>> "a.b.c.d".partition(".")
('a', '.', 'b.c.d')
```

Reply:
The idea here is whenever there is a ```data.%04.dbl.h5``` file or a ```data.%04.flt.h5``` there are corresponding files ```dbl.h5.out``` and ```flt.h5.out``` respectively which needs to be read and parsed. Also there are companion files with names ```data.%04.dbl.xmf``` and ```data.%04.flt.xmf``` which are read by the code. Since these formats don't change the string slicings were hard coded. Can you suggest anything that's better?

Reply:
> Since these formats don't change the string slicings were hard coded.

Fair point, but the reason I'm asking to avoid slicing is that it's hard to reason about for me as a mostly clueless maintainer, but also for you if you come back to this code in two years.
We could use `str.removesuffix('.h5.out')` here. This method requires Python 3.9 but I'm okay with dropping support for Python 3.8 now. Anyway this kind of technical details shouldn't clobber your PR. Can you just add a comment explaining *what 6 characters* you mean to remove, so I can refactor later with confidence ?

Reply:
> > Since these formats don't change the string slicings were hard coded.
> 
> Fair point, but the reason I'm asking to avoid slicing is that it's hard to reason about for me as a mostly clueless maintainer, but also for you if you come back to this code in two years. We could use `str.removesuffix('.h5.out')` here. This method requires Python 3.9 but I'm okay with dropping support for Python 3.8 now. Anyway this kind of technical details shouldn't clobber your PR. Can you just add a comment explaining _what 6 characters_ you mean to remove, so I can refactor later with confidence ?

Okay, I'll do that

Reply:
These are the files needed and I have added comments accordingly>
# data.%04d.<dbl/flt>.h5 -> data.%04d.<dbl/flt>.xmf
# data.%04d.<dbl/flt>.h5 -> <dbl/flt>.h5.out